packages/@uppy/core/src/index.js

```diff
@@ -1661,30 +1649,27 @@ class Uppy {
       const files = currentUpload.fileIDs.map((fileID) => this.getFile(fileID))
       const successful = files.filter((file) => !file.error)
       const failed = files.filter((file) => file.error)
-      this.addResultData(uploadID, { successful, failed, uploadID })
-    }).then(() => {
-      // Emit completion events.
-      // This is in a separate function so that the `currentUploads` variable
-      // always refers to the latest state. In the handler right above it refers
-      // to an outdated object without the `.result` property.
-      const { currentUploads } = this.getState()
-      if (!currentUploads[uploadID]) {
-        return
-      }
-      const currentUpload = currentUploads[uploadID]
-      const { result } = currentUpload
+      await this.addResultData(uploadID, { successful, failed, uploadID })
+
+      // Update currentUpload value in case it was modified asynchronously.
+      ;({ currentUploads } = this.getState())
```

Comment:
Why not keep what it used to be?

```suggestion
      const { currentUploads } = this.getState()
```

Reply:
Adding const would create a new reference, instead of updating the existing reference, which is what I want.